testsuite/integration-tests/src/test/java/org/jboss/resteasy/test/providers/sse/SseTest.java

```diff
@@ -338,6 +339,51 @@ public void testEscapedMessage() throws Exception
            Assert.assertTrue("Sent message \"" + s + "\" not found as result.", results.contains(s));
         }
      }
+    @Test
+    @InSequence(9)
+    public void testEvenSinkCloseByEvnetSource() throws Exception
+    {
+        final CountDownLatch latch = new CountDownLatch(5);
+        final List<String> results = new ArrayList<String>();
+        final AtomicInteger errors = new AtomicInteger(0);
+        Client client =  ClientBuilder.newClient();
+        WebTarget target = client.target(generateURL("/service/server-sent-events/events"));
+        SseEventSource eventSource = SseEventSource.target(target).build();
+
+        Assert.assertEquals(SseEventSourceImpl.class, eventSource.getClass());
+        eventSource.register(event -> {
+            results.add(event.readData(String.class));
+            latch.countDown();
+        }, ex -> {
+            errors.incrementAndGet();
+            ex.printStackTrace();
+            throw new RuntimeException(ex);
+        });
+        eventSource.open();
+
+        boolean waitResult = latch.await(30, TimeUnit.SECONDS);
+        Assert.assertEquals(0, errors.get());
+        Assert.assertTrue("Waiting for event to be delivered has timed out.", waitResult);
+        Client isOpenClient =  ClientBuilder.newClient();
+        Invocation.Builder isOpenRequest = isOpenClient.target(generateURL("/service/server-sent-events/isopen")).request();
+        javax.ws.rs.core.Response response = isOpenRequest.get();
+        Assert.assertFalse("EventSink isClosed=false is expected ", response.readEntity(Boolean.class));
+        
+        eventSource.close();
+        Assert.assertFalse("Closed eventSource state is expceted", eventSource.isOpen());
+        
+        boolean isClosed = false;
+        for (int i = 0 ; i < 5; i ++) 

```

Comment:
why do we need to make this 5 times? It looks suspicious to me, if this is only timing issue, it's likely to break / fail intermittently depending on the test environment.

Reply:
Server needs some time to know the connection is already closed by write another event to outputstream and get an IOException. I tried to refactor the test and remove this loop, it looks works.